### PR TITLE
Fix change return type code action for union type context

### LINKED ISSUE #36167: Use of `change return type` code action in union contexts
Refer to the following. The change return type code action doesn't work for union type contexts. In the following, what I expected to be able to do is add the type of the newly added return expression to the return type signature. i.e., make the return type a union. But instead, this changes the return type to the type of the new expression, breaking the code.

image

```ballerina
function foo() returns int {
    int|string x = 10;

    if x is int {
        return x;
    }

    return x;
}
```

### PR DESCRIPTION
**Pull request by mindula**
## Purpose
This PR adds the change return type code action for union type contexts.

Fixes #36167

## Samples
https://user-images.githubusercontent.com/61020198/171586419-e4f375d7-2224-431d-a773-c3e0f83e2de4.mov

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by TharushiJay — COMMENTED** [review]
Can we check if this could be fixed?  Return type is set as `string|function () returns boolean|int` so it gives the error `incompatible types: expected '(string|function () returns ((boolean|int)))', found 'int'`.

Screenshot 2022-06-03 at 10 16 37

File: language-server/modules/langserver-core/src/test/resources/codeaction/fix-return-type/source/fixReturnTypeInUnionContext.bal [hunk 1]
```diff
@@ -0,0 +1,10 @@
+function test() returns int {
+    int|string x = 10;
+
+    if x is int {
+        return x;
+    }
+
+    return x;
```
[the remark this answers is not in the document]
Comment (mindula) [reply language-server/modules/langserver-core/src/test/resources/codeaction/fix-return-type/source/fixReturnTypeInUnionContext.bal]: Done

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/FixReturnTypeCodeAction.java [hunk 2]
```diff
@@ -100,20 +106,62 @@ public List<CodeAction> getDiagBasedCodeActions(Diagnostic diagnostic,
...
+                types.add(Collections.singleton(funcDef.get().functionSignature().returnTypeDesc().get().type()
+                        .toString().trim().concat("|").concat("error")));
             } else {
-                types.add("error?");
+                types.add(Collections.singleton("error?"));
             }
         } else {
-            // Get all possible return types including ambiguous scenarios
-            types = CodeActionUtil.getPossibleTypes(foundType.get(), importEdits, context);
+            List<List<String>> combinations = new ArrayList<>();
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/FixReturnTypeCodeAction.java:122]:
```suggestion
            List<List<String>> combinedTypes = new ArrayList<>();
```

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/FixReturnTypeCodeAction.java [hunk 3]
```diff
@@ -100,20 +106,62 @@ public List<CodeAction> getDiagBasedCodeActions(Diagnostic diagnostic,
...
+                }
+                if (typeSymbol.get().typeKind() == TypeDescKind.FUNCTION) {
+                    combinations.add(Collections.singletonList("(" + CodeActionUtil.getPossibleTypes(typeSymbol.get(),
+                            importEdits, context).get(0) + ")"));
+                } else {
+                    combinations.add(CodeActionUtil.getPossibleTypes(typeSymbol.get(), importEdits, context));
+                }
+            }
+
+            for (List<String> stringTypes : combinations) {
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/FixReturnTypeCodeAction.java:145]:
```suggestion
            for (List<String> possibleTypes : combinations) {
```

Comment (malinthar) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/FixReturnTypeCodeAction.java:145]: Shall we extract this logic to a separate method and add comments explaining key steps

**Review by malinthar — DISMISSED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/FixReturnTypeCodeAction.java (hunk 3 above)
Comment (mindula) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/FixReturnTypeCodeAction.java]: Done

**Review by IMS94 — DISMISSED** [review]
LGTM! There's a minor suggestion.

**Review by IMS94 — APPROVED** [review]

**Review by mohanvive — APPROVED** [review]

**Pull request merged**